File: removeElement.py
from typing import List

"""
	Intuition:  traverse through the array and remove the value from array 
			    based on index i
				It is very similar to vector's erase but instead of using a for loop
				in Pynthon a while loop is easier to manipulate the i index value

	RunTime: O(N)
	Space Complexity: O(1) 

"""

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:   
        i = 0
        while (i < len(nums)):
            if (nums[i] == val):
                nums.remove(nums[i])
                i = i - 1
            i = i + 1
                
        return len(nums);

nums = [0,1,2,2,3,0,4,2]
val = 2
myObj = Solution()
answer = myObj.removeElement(nums, val)
print(answer)

File: removeElement.js
"use strict";

var removeElement = function(nums, val) 
{
    for (let i = 0; i < nums.length; i++)
    {
        if (nums[i] == val)
        {
            nums.splice(i, 1);
            i--;
        }
    }
        
    return nums.length;
};


var nums = [3,2,2,3];
var val = 3;

var answer = removeElement(nums, val);

for (let i = 0; i < nums.length; i++)
{
	console.log(nums[i]);
}

console.log("Size: " + answer);

